src/hera/workflows/_mixins.py

```diff
@@ -1013,3 +1020,29 @@ def get_artifact(self, name: str) -> Artifact:
     def get_parameter(self, name: str) -> Parameter:
         """Gets a parameter from the outputs of this subnode."""
         return self._get_parameter(name=name, subtype=self._subtype)
+
+    def get_outputs_as_arguments(self) -> List[Union[Parameter, Artifact]]:
+        """Get all output parameters and artifacts as a combined list from this task/step for use as arguments.
+
+        This is useful for when all the inputs of another template match all the outputs of this template. It
+        is also possible to combine the outputs of multiple templates if they collectively match the inputs of
+        another template.
+        """
+        if isinstance(self.template, str):
+            raise ValueError(f"Cannot get outputs when the template was set via a name: {self.template}")
+
+        # here, we build the template early to verify that we can get the outputs
+        if isinstance(self.template, Templatable):
+            template = self.template._build_template()
+        else:
+            template = self.template
+
+        # at this point, we know that the template is a `Template` object
+        assert isinstance(template, Template)
+
+        if template.outputs is None:
+            raise ValueError(f"Template '{template.name}' has no outputs")
+
+        parameters = [self.get_parameter(p.name) for p in template.outputs.parameters or []]
+        artifacts = [self.get_artifact(art.name) for art in template.outputs.artifacts or []]
+        return parameters + artifacts
```

Comment:
It still looks like we sometimes return an empty list for no parameters and sometimes raise, depending on whether outputs is missing or empty. I'd prefer we pick one and be consistent. Up to you though.